xlml/utils/metric.py

```diff
@@ -611,19 +611,30 @@ def process_metrics(
 
   # process metrics, metadata, and profile
   if task_metric_config:
-    if task_metric_config.json_lines:
-      metric_history_rows_list, metadata_history_rows_list = process_json_lines(
-          base_id, task_metric_config.json_lines.file_location
-      )
-    elif task_metric_config.use_runtime_generated_filename:
-      metric_history_rows_list, metadata_history_rows_list = process_json_lines(
-          base_id, file_location
-      )
+    if task_metric_config.json_lines.file_location:
+      if task_metric_config.use_runtime_generated_gcs_folder:
+        metric_history_rows_list, metadata_history_rows_list = process_json_lines(
+            base_id,
+            os.path.join(folder_location, task_metric_config.json_lines.file_location),
```

Comment:
The only part that differs in these cases is the file path, right? Can you simplify the cases to something like:

```
absolute_path = os.path.join(folder_location, task_metric_config.json_lines.file_location)  if use_runtime_generated_gcs_folder else task_metric_config.json_lines.file_location
... = process_json_lines(base_id, absolute_path)
```

IMO we should start standardizing on use_runtime_generated_gcs_folder and eliminate usage of `task_metric_config.json_lines.file_location` as an absolute path

Reply:
Sounds good, simplified the logic.